webhooks/cluster.go

```diff
@@ -92,7 +118,21 @@ func (c *Cluster) ValidateUpdate(_ context.Context, oldObj, newObj runtime.Objec
 	// The idea is to add new validation which requires a reader and Cluster variable/patch validation
 	// in the webhook package.
 	// When we drop the method in the API package we must inline it here.
-	return cluster.ValidateUpdate(oldCluster)
+	if err := cluster.ValidateUpdate(oldCluster); err != nil {
+		return err
+	}
+
+	// Additional dynamic checks are done if ClusterTopology is enabled and the Cluster has an associated Topology.
+	if feature.Gates.Enabled(feature.ClusterTopology) && cluster.Spec.Topology != nil {
+		if err := c.checkClusterClassExists(ctx, cluster.Namespace, cluster.Spec.Topology.Class); err != nil {
```

Comment:
Should we execute this test only if class name is changed, thus avoiding using the client if not required?

Reply:
Good idea!

Reply:
Up to you, but that optimization will probably won't work long as we need the ClusterClass to validate the variables on the Cluster with one of the next PRs

Reply:
I'll leave this out for now. There's a few interdependencies in the checks so it might be better to leave optimization like this until more or most of the checks are in place.